dawn/test/unit-test/dawn/Optimizer/Passes/TestPassRemoveScalars.cpp

```diff
@@ -0,0 +1,587 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Optimizer/PassRemoveScalars.h"
+#include "dawn/Unittest/IIRBuilder.h"
+
+#include <gtest/gtest.h>
+
+using namespace dawn;
+
+namespace {
+
+std::shared_ptr<iir::Expr> getRhsOfAssignment(const std::shared_ptr<iir::Stmt> stmt) {
+  if(stmt->getKind() == iir::Stmt::Kind::VarDeclStmt) {
+    const auto& varDeclStmt = std::dynamic_pointer_cast<iir::VarDeclStmt>(stmt);
+    return varDeclStmt->getInitList()[0];
+  } else if(stmt->getKind() == iir::Stmt::Kind::ExprStmt) {
+    const auto& exprStmt = std::dynamic_pointer_cast<iir::ExprStmt>(stmt);
+    if(exprStmt->getExpr()->getKind() == iir::Expr::Kind::AssignmentExpr) {
+      const auto& assignmentExpr =
+          std::dynamic_pointer_cast<iir::AssignmentExpr>(exprStmt->getExpr());
+      return assignmentExpr->getRight();
+    }
+  }
+
+  return nullptr;
+}
+
+iir::DoMethod& getDoMethod(std::shared_ptr<iir::StencilInstantiation>& si) {
+  auto& iir = si->getIIR();
+  auto& stencil = iir->getChild(0);
+  auto& ms = stencil->getChild(0);
+  auto& stage = ms->getChild(0);
+  return *stage->getChild(0);
+}
+
+std::shared_ptr<iir::Stmt> getNthStmt(std::shared_ptr<iir::StencilInstantiation>& si, int n) {
```

Comment:
should something like this not live in unittest utils?

Reply:
Moving this to utils